src/mamba/py_interface.cpp

```diff
@@ -246,6 +254,82 @@ PYBIND11_MODULE(mamba_api, m)
         .def(py::init<const std::string&, const std::string&, const std::string&, std::size_t>())
         .def_readwrite("name", &PackageInfo::name);
 
+    // Content trust - Package signature and verification
+    m.def("generate_ed25519_keypair", &validate::generate_ed25519_keypair_hex);
+    m.def(
+        "sign",
+        [](const std::string& data, const std::string& sk) {
+            std::string signature;
+            if (!validate::sign(data, sk, signature))
+                throw std::runtime_error("Signing failed");
+            return signature;
+        },
+        py::arg("data"),
+        py::arg("secret_key"));
+
+    py::class_<validate::Key>(m, "Key")
+        .def_readwrite("keytype", &validate::Key::keytype)
+        .def_readwrite("scheme", &validate::Key::scheme)
+        .def_readwrite("keyval", &validate::Key::keyval)
+        .def("to_json",
```

Comment:
Do we really need `pybind11_json`? Because I would expect these functions to return a `std::string` that contains the JSON.

Reply:
I see, in fact I was also using it in `RootImpl`, `KeyMgrRole` and `PkgMgrRole` constructors.
It was pretty handy in `Quetz` but I can try to do another way to remove this extra dep

Reply:
why not just:

```
std::stringstream out;
out << j.dump(4);
return out.str();
```

Reply:
maybe I am not understanding correctly what pybind json does? I understand it creates an object out of the string?

But do we need that given that we already access variables on the `Key` object through `.def_readwrite...`?

Reply:
for `Key` it's definitely because I was lazy ;)
for the main content roles, I just created factories to have Python constructor taking json strings instead of either `fs::path` or `json` types. The factory allows to disambiguate the signature between `std::string` and `fs::path`. Maybe there is also another way to do it.

Reply:
I didn't needed the factory pattern, just added a new constructor